pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -233,7 +233,7 @@ def _check_consider_using_dict_items(self, node: astroid.For) -> None:
                     and subscript == subscript.parent.target
                 ):
                     # Ignore this subscript if it is the target of an assignment
-                    continue
+                    return  # Early termination as dict index lookup is necessary
```

Comment:
```suggestion
                    # Early termination as dict index lookup is necessary
                    return
```
Probably just my personal preference, but what do you think?

Reply:
fixed in 14b6db0